### PR TITLE
Add exception handling to os.remove in the remove-asset-by_pathfang method of the avocado/utilities/asset.py file

### PR DESCRIPTION
**Pull request by faker-king**
Using pathlib can make file path operations more object-oriented and easy to understand, and have better compatibility across different operating systems.

### CODE REVIEW

**Comment by faker-king** [comment]
hi @richtja, Can you help me see if I can merge the code

File: avocado/utils/asset.py [hunk 1]
```diff
@@ -684,9 +685,15 @@ def remove_asset_by_path(cls, asset_path):
 
         :param asset_path: full path of the asset file.
         """
-        os.remove(asset_path)
-        filename = f"{asset_path}-CHECKSUM"
-        os.remove(filename)
```
Comment (richtja) [inline avocado/utils/asset.py:689]: I would keep the `os.remove`. I don't see any additional benefit with usage of `pathlib`, especially when it is inconsistent with other parts of asset utility which uses  `os.remove`.

File: avocado/utils/asset.py [hunk 2]
```diff
@@ -684,9 +685,15 @@ def remove_asset_by_path(cls, asset_path):
...
+        try:
+            file_path = pathlib.Path(asset_path)
+            file_path.unlink()
+            checksum_path = pathlib.Path(f"{asset_path}-CHECKSUM")
+            checksum_path.unlink()
+        except FileNotFoundError:
```
Comment (richtja) [inline avocado/utils/asset.py:693]: The exception handling is a great idea, and it is something which is needed here. Good job.

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @faker-king please remove 9806712b4ec63399c72f7ca243f005aa00301637 and 1051bc0ec08f27be2dab31f7384053faadfc8898. IMO, they are not related to this PR. To be honest, I am not sure about usage of `pathlib` in one specific method of `asset` utility. Correct me if I am wrong, but I don't see any additional benefit in this change.

But you catch that we don't have exception handling for this file manipulation, and that would be a great update. Therefore, please add only the exception handling. Thanks

**Comment by faker-king** [comment]
Hi @richtja, I am very sorry for the delay in modifying this PR. I have already made the necessary changes as per your request. Could you please double check.

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @faker-king, thank you for your updates.  The overall changes LGTM, but you still have 9806712b4ec63399c72f7ca243f005aa00301637 and 1051bc0ec08f27be2dab31f7384053faadfc8898 commit which are not related to this PR, please drop those.

 Also, can you please update the title of ee32a501a4cd106bf13019bec5d5b721da379c91, from your description you are updating the exception handling to all os.remove calls, but this commit is only about `avocado/utils/asset.py`. Please be more specific. Thank you.

**Comment by faker-king** [comment]
@richtja Due to some permission issues with my account, it was verified and I have made the necessary modifications as per your suggestion to see if it can be merged

**Comment by richtja** [comment]
Hi @faker-king, thank you for your updates. I'm still missing the commit title update, your changes don't add exception handling to os.remove in the whole avocado project, it is located only to avocado/utils/asset.py. Can you please update the title to be more specific? Thank you.

**Comment by faker-king** [comment]
@richtja Modified

**Comment by richtja** [comment]
> @richtja Modified

Hi @faker-king, sorry but maybe I haven't expressed myself clearly, I was talking about commit headline. It is needed for easier search thought changes in commit history in the future. Can you please update the [commit headline](https://github.com/avocado-framework/avocado/pull/5979/commits/ae3d836fc7f8b42e0488e7712a06de211f1b2691) as well, thanks.

**Comment by faker-king** [comment]
@richtja Haha, it's okay, I didn't understand either,  it has been modified.

**Review by richtja — APPROVED** [review]
Hi @faker-king, it LGTM now. Thanks for your changes.

**Pull request merged**